src/Functions.hs

```diff
@@ -7,13 +7,15 @@ import Utils
 import Debug.Trace
 import qualified Data.Map.Strict as M
 
-type ObjFnOn a = [Obj' a] -> [a] -> a
-type ConstrFnOn a = [Obj' a] -> [a] -> a
-type ObjFn = forall a. (Autofloat a) => [Obj' a] -> [a] -> a
-type ConstrFn = forall a. (Autofloat a) => [Obj' a] -> [a] -> a
-type ObjFnInfo a = (ObjFnOn a, Weight a, [Name], [a])
-type ConstrFnInfo a = (ConstrFnOn a, Weight a, [Name], [a])
-type Weight a = a
+type ObjFnOn    a = [Obj' a] -> [TypeIn a] -> a
+type ConstrFnOn a = [Obj' a] -> [TypeIn a] -> a
+type ObjFn    = forall a. (Autofloat a) => [Obj' a] -> [TypeIn a] -> a
```

Comment:
I'm concerned about whether objfns get arguments in the right order as specified in Style. Can we document how the ordering is preserved or not preserved?

Reply:
Noted in documentation of `applyAndSet`